### PR TITLE
refactor: Rename autogen

### PR DESCRIPTION
**Pull request by webbdays**
**Summary:**  
rename autogen

**Issue Reference(s):**
[comment](https://github.com/tailcallhq/tailcall/pull/1704#issuecomment-2050001951)

**Build & Testing:**

- [yes ] I ran `cargo test` successfully.
- [ ] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [ ] I have added relevant unit & integration tests.
- [ ] I have updated the [documentation] accordingly.
- [ yes] I have performed a self-review of my code.
- [ yes] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

## Summary by CodeRabbit

- **Refactor**
	- Updated internal package references to use the new `tailcall-autogen` naming convention.

### CODE REVIEW

File: lint.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ run_prettier() {
 
 run_autogen_schema() {
     MODE=$1
-    cargo run -p autogen $MODE
+    cargo run -p tailcall-autogen $MODE
```
[the remark this answers is not in the document]
Comment (webbdays) [reply lint.sh:39]: example.

[the remark this answers is not in the document]
Comment (webbdays) [reply lint.sh:39]: example to justify to your comments

[the remark this answers is not in the document]
Comment (webbdays) [reply lint.sh:39]: @tusharmath
its not that required.
but good to have

**Review by tusharmath — APPROVED** [review]

**Comment by tusharmath** [comment]
@webbdays conflicts.

File: lint.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (webbdays) [reply lint.sh:39]: not a good time to do it.

[the remark this answers is not in the document]
Comment (webbdays) [reply lint.sh:39]: not a good time to do it.

**Comment by webbdays** [comment]
@tusharmath 
done

**Pull request merged**